apps/studio/src/server/modules/folder/__tests__/folder.router.test.ts

```diff
@@ -0,0 +1,634 @@
+  })
+
+  describe("getMetadata", () => {
+    it("should throw 401 if not logged in", async () => {
+      // Act
+      const result = unauthedCaller.getMetadata({
+        siteId: 1,
+        resourceId: -1,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "UNAUTHORIZED" }),
+      )
+    })
+
+    it("should throw 404 if `siteId` does not exist", async () => {
+      // Arrange
+      const invalidSiteId = 999
+      const { site } = await setupSite()
+      await setupAdminPermissions({
+        userId: session.userId,
+        siteId: site.id,
+      })
+      expect(site.id).not.toEqual(invalidSiteId)
+
+      // Act
+      const result = caller.getMetadata({
+        siteId: invalidSiteId,
+        resourceId: 1,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "You do not have sufficient permissions to perform this action",
+        }),
+      )
+    })
+
+    it("should throw 404 if `folderId` does not exist", async () => {
+      // Arrange
+      const { site } = await setupSite()
+      await setupAdminPermissions({
+        userId: session.userId,
+        siteId: site.id,
+      })
+
+      // Act
+      const result = caller.getMetadata({
+        siteId: site.id,
+        resourceId: 999,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message: "This folder does not exist",
+        }),
+      )
+    })
+
+    it("should throw 403 if user does not write access to the site", async () => {
+      // Arrange
+      const { folder, site } = await setupFolder()
+
+      // Act
+      const result = caller.getMetadata({
+        siteId: site.id,
+        resourceId: Number(folder.id),
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "You do not have sufficient permissions to perform this action",
+        }),
+      )
+    })
+
+    it("should return 200 ", async () => {
+      // Arrange
+      const { folder, site } = await setupFolder()
+      await setupAdminPermissions({ userId: session.userId, siteId: site.id })
+
+      // Act
+      const result = await caller.getMetadata({
+        siteId: site.id,
+        resourceId: Number(folder.id),
+      })
+
+      // Assert
+      const expected = await db
+        .selectFrom("Resource")
+        .select(["Resource.title", "Resource.permalink", "Resource.parentId"])
+        .where("id", "=", folder.id)
+        .executeTakeFirst()
+      expect(result).toEqual(expected)
+    })
+  })
+
+  describe("editFolder", () => {
+    afterEach(async () => {
+      await clearPermissions()
+    })
+
+    it("should return an empty array if the resourceId is not a folder", async () => {
```

Comment:
test is correct, but should be "undefined" instead of "empty array"

slight nit - can move this after the error tests to be consistent? thanks!